### PR TITLE
DoubleS1Scatter Tf nn switch

### PR DESCRIPTION
**Pull request by ctherreau**
Update CorrectedDoubleS1Scatter to include TensorFlow based position in the same way as in corrections minitree

### CODE REVIEW

**Comment by tunnell** [comment]
Flake8 detected 38 issues on 85bcd82e25deca1dd67798957dd28502678e4336
Visit https://sider.review/gh/repos/50927571/pulls/246 to review the issues.

<sub>posted by Sider</sub>

**Comment by tunnell** [comment]
Flake8 detected 8 issues on 6077b3141d8dcbb2beeba4a4b1dcd6b5cc8cf5b4
Visit https://sider.review/gh/repos/50927571/pulls/246 to review the issues.

<sub>posted by Sider</sub>

File: hax/treemakers/CorrectedDoubleS1Scatter.py [hunk 1]
```diff
@@ -217,131 +176,153 @@ def extract_data(self, event):
...
-        result['s1_int_b_xyz_true_correction_nn_fdc_3d'] = (
-            1 / self.corrections_handler.get_correction_from_map(
-                "s1_corrected_lce_map_nn_fdc_3d", self.run_number, cvals)
-        )
-        result['cs1_b_int_b'] = peaks[s1_b].area * result['s1_int_b_xyz_true_correction_nn_fdc_3d']
+        # Correction only with int_a
+        int_s_default = 'int_a_'
+        for algo in ['nn_tf', 'nn', 'tpf']:
+            # Correct S2
+            result[int_s_default+'r_observed_' + algo] = np.sqrt(result[int_s_default+'x_observed_' + algo] ** 2
```
Comment (tunnell) [inline hax/treemakers/CorrectedDoubleS1Scatter.py]: __[Flake8]__ [W291] trailing whitespace

<sub>posted by Sider</sub>

**Comment by ctherreau** [comment]
Some plots to check; one Kr83m run from SR1 (170206_1255); no cut
 - cS1_a_nn_tf vs cS2_a_nn_tf 
 - int_a_r_3d_nn_tf vs int_a_z_3d_nn_tf

cs1cs2
pos_rec

File: hax/treemakers/CorrectedDoubleS1Scatter.py [hunk 2]
```diff
@@ -3,76 +3,41 @@
...
-
     ### DoubleScatter Specific Output:
     - ds_s1_b_n_distinct_channels: number of PMTs contributing to s1_b distinct from the PMTs that contributed to s1_a
     - ds_s1_dt : delay time between s1_a_center_time and s1_b_center_time
     - ds_second_s2: 1 if selected interactions have distinct s2s
+    ### Position Corrections : Same as corrections minitree but for int_a and int_b position
+    ### Signal corrections: Same as corrections minitree using int_a positions for position dependant corrections
     """
-    __version__ = '2.0'
+    __version__ = '2.1.2'
```
Comment (feigaodm) [inline hax/treemakers/CorrectedDoubleS1Scatter.py]: Use 2.1 instead of 2.1.2

File: hax/treemakers/CorrectedDoubleS1Scatter.py [hunk 3]
```diff
@@ -217,131 +176,153 @@ def extract_data(self, event):
...
-                                            self.corrections_handler.get_correction_from_map(
-                                                "s2_xy_map", self.run_number, cvals))
-        result['s2_a_xy_correction_top'] = (1.0 /
-                                            self.corrections_handler.get_correction_from_map(
-                                                "s2_xy_map", self.run_number, cvals, map_name='map_top'))
-        result['s2_a_xy_correction_bottom'] = (1.0 /
-                                               self.corrections_handler.get_correction_from_map(
-                                                   "s2_xy_map", self.run_number, cvals, map_name='map_bottom'))
-
+        int_signal = ['int_a_', 'int_b_']
```
Comment (feigaodm) [inline hax/treemakers/CorrectedDoubleS1Scatter.py]: ideally you can use this for loop for above nn&tpf positions, but it's fine as is.

File: hax/treemakers/CorrectedDoubleS1Scatter.py (hunk 2 above)
Comment (ctherreau) [reply hax/treemakers/CorrectedDoubleS1Scatter.py]: ok

**Comment by l-althueser** [comment]
> Some plots to check; one Kr83m run from SR1 (170206_1255); no cut
> 
> cS1_a_nn_tf vs cS2_a_nn_tf
> int_a_r_3d_nn_tf vs int_a_z_3d_nn_tf

To my knowledge, we shouldn't have plots directly in GitHub (as stated in the PR templates). They can be provided by a wiki note if needed.

**Comment by feigaodm** [comment]
thanks @l-althueser , I forgot about this rule for calibration data. Maybe @ctherreau can post it somewhere else and link them. Without plots, it's really hard to review.

**Comment by ctherreau** [comment]
> > Some plots to check; one Kr83m run from SR1 (170206_1255); no cut
> > cS1_a_nn_tf vs cS2_a_nn_tf
> > int_a_r_3d_nn_tf vs int_a_z_3d_nn_tf
> 
> To my knowledge, we shouldn't have plots directly in GitHub (as stated in the PR templates). They can be provided by a wiki note if needed.

Thanks @l-althueser, I change my comment with a wiki link

**Pull request merged**